### ISSUE TITLE
Concerns about the initial number of 3D Gaussians

### ISSUE THREAD

**Issue by Time-Lord12th** [donydchen/mvsplat#8]
Hi, thanks for the great work. The paper mentions D=128, and "After obtaining the multi-view depth predictions, we directly unproject them to 3D point clouds using the camera parameters." 
So is the initial number of 3D Gaussians 128*K? Isn't this number too small? Approximately how many 3D Gaussians are there after the training is completed?

**Comment by donydchen** [comment]
Hi @Time-Lord12th, thanks for your interest. The number of 3D Gaussians is $H \times W \times D \times N$, where $H, W$ are the image resolutions, $D=128$ is the number of depth planes, and $N$ is the number of context/input views. The number of 3D Gaussians is fixed; we only update the properties of the Gaussians but not the quantity. And we find that such a number is enough for our experimental datasets.

**Comment by Time-Lord12th** [comment]
Thanks for your fast and detailed reply!

**Comment by Time-Lord12th** [comment]
Sorry to bother you again. The number of 3D Gaussians should be HxWxN? As stated in the paper：
![image](https://github.com/donydchen/mvsplat/assets/53140195/42c1b040-9f27-4c51-bef4-4ddde9db13a2)


Could I possibly ask another question? For ablations, when removing `cost volume`, how does MVSplat predict 3D Gaussian parameters, does it predict probabilistic depth like pixelSplat?
Thanks!

**Comment by donydchen** [comment]
Hi @Time-Lord12th. Yes, you are correct. Sorry for the mistake I made above. It is pixel-aligned and should be just $H \times W \times N$. The depth dimension will be reduced to 1 via weighted average using the weights obtained via softmax, implementation details are [here](https://github.com/donydchen/mvsplat/blob/main/src/model/encoder/costvolume/depth_predictor_multiview.py#L331-L336).

--- 

When removing cost volume, we simply rely on the feature to regress the Gaussian centers, _without_ using any probabilistic sampling like pixelSplat. Related implementation details are [here](https://github.com/donydchen/mvsplat/blob/main/src/model/encoder/costvolume/depth_predictor_multiview.py#L293-L295).

---

Feel free to ask any questions regarding our MVSplat; we would be more than happy to answer them. Thanks.
